### PR TITLE
ci: changed workflow to infra's v2.0.0

### PR DESCRIPTION
**Pull request by nitzanmorr**
Since pull_request.yaml@postgres-service-in-pr doesn't exist anymore, I changed the shared workflow tag to v2 and enabled postgres deployment on pull-request workflow.
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: .github/workflows/build_and_push.yaml [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
-      - run: docker build . -t ${{ env.Docker_Repository }}
-      - run: docker push ${{ env.Docker_Repository }}
\ No newline at end of file
+  build_and_push_docker:
+      uses: MapColonies/shared-workflows/.github/workflows/build-and-push-docker.yaml@v2
+      secrets: inherit
+  
+  build_and_push_helm:
+      uses: MapColonies/shared-workflows/.github/workflows/build-and-push-helm.yaml@v2
+      secrets: inherit
```
Comment (netanelC) [inline .github/workflows/build_and_push.yaml:27]: Add empty line

**Review by netanelC — CHANGES_REQUESTED** [review]
Good Job!

File: .github/workflows/build_and_push.yaml (hunk 1 above)
Comment (nitzanmorr) [reply .github/workflows/build_and_push.yaml]: fixed

Comment (netanelC) [reply .github/workflows/build_and_push.yaml]: Remove the spaces from the empty line

Comment (nitzanmorr) [reply .github/workflows/build_and_push.yaml]: removed

**Review by netanelC — APPROVED** [review]

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Pull request merged**